servant-client/src/Servant/Client/PerformRequest/GHCJS.hs

```diff
@@ -62,7 +62,9 @@ performXhr xhr request = do
   callback <- onReadyStateChange xhr $ do
     state <- readyState xhr
     case state of
-      4 -> putMVar waiter ()
+      4 -> do
+          toPut <- isEmptyMVar waiter
```

Comment:
A Race condition can still happen between this and `putMVar`. It won't actually fix the problem I think.


I think `tryPutMVar` is more appropriate. It's an atomic version of this behaviour.


Also, is it really a problem in the first place? I can not see a situation here where `putMVar` will actually block as `waiter` will always be empty before `putMVar` is called.   (Because there is only one `putMVar` happening, namely on the `onReadyStateChange` callback)